### PR TITLE
Update GUI documentation

### PR DESCRIPTION
**Pull request by ben-dale**
This adds a load more content to the docs, including screenshots, specifically around the GUI components.

All of the code I wrote for these docs was ran within the Lanterna project, meaning that some of the code might not be compatible with the latest Lanterna 3 Beta version. Is there going to be a new Beta release soon? The last one was released at the end of June.

### CODE REVIEW

**Comment by mabe02** [comment]
Yes, new beta soon!

File: src/test/java/com/googlecode/lanterna/gui2/ComboBoxTest.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ public static void main(String[] args) throws IOException, InterruptedException
 
     @Override
     public void init(WindowBasedTextGUI textGUI) {
-        final BasicWindow window = new BasicWindow("ComboBoxTest");
+        final BasicWindow window = new BasicWindow();
```
Comment (avl42) [inline src/test/java/com/googlecode/lanterna/gui2/ComboBoxTest.java]: Were these two changes intended, or did they slip in?

It seems to me as if some features are being *removed* here...
 src/test/java/com/googlecode/lanterna/gui2/ComboBoxTest.java  and src/test/java/com/googlecode/lanterna/gui2/ListBoxTest.java

**Comment by ben-dale** [comment]
Woops! Those shouldn't have gone in! My bad.

I'll bring up a PR now which reverts their changes

File: src/test/java/com/googlecode/lanterna/gui2/ComboBoxTest.java (hunk 1 above)
Comment (ben-dale) [inline src/test/java/com/googlecode/lanterna/gui2/ComboBoxTest.java]: Woops! Those shouldn't have gone in! My bad.

I'll bring up a PR now which reverts their changes

Comment (avl42) [inline src/test/java/com/googlecode/lanterna/gui2/ComboBoxTest.java]: Anyway, thanks a lot for your valuable contribution to the documentation!

**Pull request merged**